### ISSUE TITLE
kubelet cannot show Docker-CE version correctly 

### ISSUE THREAD

**Issue by dixudx** [kubernetes/kubernetes#54039]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
1. Install Docker-CE

```
$ docker version
Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:42:18 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:40:56 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

2. Bootstrap K8s cluster normally
3. when getting nodes,

```
$ kubectl get node -o wide
NAME        STATUS    ROLES     AGE       VERSION   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION     CONTAINER-RUNTIME
server-02   Ready     master    1d        v1.8.1    <none>        Ubuntu 16.04.2 LTS   4.4.0-83-generic   docker://Unknown
```

**What you expected to happen**:
Should show a version intead of `docker://Unknown`, which is confusing.

**How to reproduce it (as minimally and precisely as possible)**:
As above

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.1", GitCommit:"f38e43b221d08850172a9a4ea785a86a3ffa3b3a", GitTreeState:"clean", BuildDate:"2017-10-11T23:27:35Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.1", GitCommit:"f38e43b221d08850172a9a4ea785a86a3ffa3b3a", GitTreeState:"clean", BuildDate:"2017-10-11T23:16:41Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by xyalan** [comment]
I have same problem in v1.8.3+coreos.0.

**Comment by dixudx** [comment]
@xyalan Currently the fix is in v1.9 and master branch. Not available for v1.8.

@wojtek-t Shall we cherry-pick to v1.8?

**Comment by wojtek-t** [comment]
@jpbetz is patch release manager for v1.8

**Comment by Bobi-zhou** [comment]
I have same problem in v1.8.4 too，how to fix it？

**Comment by dixudx** [comment]
@xyalan @Bobi-zhou Sorry for the delay. I've already cherry-picked the fix to release-1.8. Please refer to #56945. The fix should be available in v1.8.6.

For current already-released v1.8 version, please cherry-pick the fix manually if you really need it and re-compile `kubelet`.

And v1.9 is coming out soon, targeted on Dec 13. You can wait for this new version, which has already included the fix.
